### ISSUE TITLE
Resume from last checkpoint 

### ISSUE THREAD

**Issue by haschel** [tijiang13/InstantAvatar#11]
When I run run-demo.sh
The prompt ask me for a resume from last checkpoint. 

What I have to do?

bash ./bash/run-demo.sh
Global seed set to 42
Switch to /home/arfuture/Documentos/Avatares/InstantAvatar/outputs/peoplesnapshot/demo/male-3-casual
[train] Loading from /home/arfuture/Documentos/Avatares/InstantAvatar/data/PeopleSnapshot/male-3-casual/poses/anim_nerf_train.npz
[val] Loading from /home/arfuture/Documentos/Avatares/InstantAvatar/data/PeopleSnapshot/male-3-casual/poses/anim_nerf_val.npz
[test] Loading from /home/arfuture/Documentos/Avatares/InstantAvatar/data/PeopleSnapshot/male-3-casual/poses/anim_nerf_test.npz
[2023-05-10 08:55:47,109][torch.distributed.nn.jit.instantiator][INFO] - Created a temporary directory at /tmp/tmpzf4rgkye
[2023-05-10 08:55:47,110][torch.distributed.nn.jit.instantiator][INFO] - Writing /tmp/tmpzf4rgkye/_remote_module_non_scriptable.py
/home/arfuture/anaconda3/envs/env/lib/python3.8/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
/home/arfuture/anaconda3/envs/env/lib/python3.8/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=VGG16_Weights.IMAGENET1K_V1`. You can also use `weights=VGG16_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
/home/arfuture/anaconda3/envs/env/lib/python3.8/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
/home/arfuture/anaconda3/envs/env/lib/python3.8/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=AlexNet_Weights.IMAGENET1K_V1`. You can also use `weights=AlexNet_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
Resume from checkpoints/last.ckpt
Restoring states from the checkpoint path at checkpoints/last.ckpt
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
/home/arfuture/anaconda3/envs/env/lib/python3.8/site-packages/pytorch_lightning/trainer/connectors/checkpoint_connector.py:247: UserWarning: You're resuming from a checkpoint that ended mid-epoch. Training will start from the beginning of the next epoch. This can cause unreliable results if further training is done, consider using an end of epoch checkpoint.
  rank_zero_warn(
Restored all states from the checkpoint file at checkpoints/last.ckpt

  | Name       | Type       | Params
------------------------------------------
0 | net_coarse | NeRFNGPNet | 13.0 M
1 | loss_fn    | NGPLoss    | 14.7 M
2 | evaluator  | Evaluator  | 2.5 M 
------------------------------------------
13.0 M    Trainable params
17.2 M    Non-trainable params
30.2 M    Total params
120.893   Total estimated model params size (MB)
/home/arfuture/anaconda3/envs/env/lib/python3.8/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:631: UserWarning: Checkpoint directory /home/arfuture/Documentos/Avatares/InstantAvatar/outputs/peoplesnapshot/demo/male-3-casual/checkpoints exists and is not empty.
  rank_zero_warn(f"Checkpoint directory {dirpath} exists and is not empty.")
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
Testing: 100%|██████████████████████████████████| 55/55 [00:06<00:00,  8.22it/s]
Global seed set to 42
Switch to /home/arfuture/Documentos/Avatares/InstantAvatar/outputs/peoplesnapshot/demo/male-3-casual
[train] Loading from /home/arfuture/Documentos/Avatares/InstantAvatar/data/PeopleSnapshot/male-3-casual/poses/anim_nerf_train.npz
[val] Loading from /home/arfuture/Documentos/Avatares/InstantAvatar/data/PeopleSnapshot/male-3-casual/poses/anim_nerf_val.npz
[test] Loading from /home/arfuture/Documentos/Avatares/InstantAvatar/data/PeopleSnapshot/male-3-casual/poses/anim_nerf_test.npz
[2023-05-10 08:55:58,591][torch.distributed.nn.jit.instantiator][INFO] - Created a temporary directory at /tmp/tmpr6vxqn_h
[2023-05-10 08:55:58,591][torch.distributed.nn.jit.instantiator][INFO] - Writing /tmp/tmpr6vxqn_h/_remote_module_non_scriptable.py
/home/arfuture/anaconda3/envs/env/lib/python3.8/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
/home/arfuture/anaconda3/envs/env/lib/python3.8/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=VGG16_Weights.IMAGENET1K_V1`. You can also use `weights=VGG16_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
/home/arfuture/anaconda3/envs/env/lib/python3.8/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
/home/arfuture/anaconda3/envs/env/lib/python3.8/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=AlexNet_Weights.IMAGENET1K_V1`. You can also use `weights=AlexNet_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
Resume from checkpoints/last.ckpt
320it [00:23, 13.51it/s]

**Comment by tijiang13** [comment]
Hi,

From the error msg, I didn't see where it asks you "for a resume from last checkpoint." As far as I can see, the last line of the log says it loads the checkpoint automatically.

Best,
Tianjian

**Comment by haschel** [comment]
So what I need to do for visualize the 3d avatar model and the animation generated?

Thanks for all!!

specially for your great work and for your quick response.

congratulations!!

**Comment by tijiang13** [comment]
> So what I need to do for visualize the 3d avatar model and the animation generated?
> 
> Thanks for all!!
> 
> specially for your great work and for your quick response.
> 
> congratulations!!

Btw, the visualization can be found in the `outputs/peoplesnapshot/demo/` folder.

Best,
Tianjian
